gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```diff
@@ -205,6 +209,12 @@ protected void runWorkUnits(List<WorkUnit> workUnits) throws Exception {
       this.eventSubmitter.submit(countEventBuilder);
       LOGGER.info("Emitting WorkUnitsCreated Count: " + countEventBuilder.getCount());
 
+      MetricContext
+          metricContext = Instrumented.getMetricContext(ConfigUtils.configToState(ConfigFactory.empty()), getClass());
+      String workUnitCountName = MetricRegistry.name(this.jobContext.getJobState().getProp(ConfigurationKeys.FLOW_GROUP_KEY),
+          this.jobContext.getJobState().getProp(ConfigurationKeys.FLOW_NAME_KEY), "workunits");
+      metricContext.newContextAwareGauge(workUnitCountName, () -> workUnits.size());
```

Comment:
You need to register this gauge to the metricContext, right?

Reply:
Yes thank you, registered the metric. Here is the [PR](https://github.com/apache/gobblin/commit/6bed94a0f213ef64b225bb05b80ab338c25bbc17) relating to the register requirement for future reference.